tests/fs_utf8.rs

```diff
@@ -1504,11 +1504,15 @@ fn create_dir_long_paths() {
     path.extend(iter::repeat(OsStr::new("a")).take(PATH_LEN - utf16_len));
 
     // This should succeed.
-    tmpdir.create_dir(&path).unwrap();
+    tmpdir
+        .create_dir(&PathBuf::from_path_buf(path.clone().into()).unwrap())
```

Comment:
(I try to heavily use camino where it makes sense)

Reply:
This particular diff is awkward because it's ported from a std test which eagerly converts the path into an `OsString`, and I attempted to port it to fs_utf8 with minimal changes. In normal usage, once you have a `Utf8PathBuf`, you'd just keep it as such, and not convert it to `OsString` :-).